sdks/python/apache_beam/runners/interactive/utils.py

```diff
@@ -142,3 +142,61 @@ def obfuscate(*inputs):
   str_inputs = [str(input) for input in inputs]
   merged_inputs = '_'.join(str_inputs)
   return hashlib.md5(merged_inputs.encode('utf-8')).hexdigest()
+
+
+class ProgressIndicator(object):
+  """An indicator visualizing code execution in progress."""
+  spinner_template = """
+            <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
```

Comment:
Is it fine to link to a 3rd party domain here?

Reply:
Yes, it's just some CSS file. We've been using the same CSS file in the IPythonLogHandler too. Bootstrap has been widely known and used, so it's very safe.

Reply:
This will ping a third party server from users environment. I do not think this is great, but if we do not have a better option, that is fine.

Reply:
Added a TODO item to change the CDN. And once we create some Jupyter Labextension, we could have HTML+JS+CSS dependencies pre-installed, then the CDN would not be necessary.